docs/guides/src/main/server/vault.adoc

```diff
@@ -0,0 +1,59 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+
+<@tmpl.guide
+title="Using Kubernetes Secrets"
+summary="Learn how to use Kubernetes / OpenShift secrets in Keycloak"
+priority=30
+includedOptions="vault vault-*">
+
+Keycloak supports a file based vault implementation for Kubernetes / OpenShift secrets. Mount Kubernetes secrets into the Keycloak Container, and the data fields will be available in the mounted folder with a flat-file structure.
```

Comment:
Should we cover environment variables approach as well? From memory I think it supports either right?